### PR TITLE
feat(react-query): add abortOnUnmount to TRPC Client Options

### PR DESCRIPTION
**Pull request by bautistaaa**
Closes https://github.com/trpc/trpc/issues/4040

## 🎯 Changes

What changes are made in this PR? Is it a feature or a bug fix?

## ✅ Checklist

- [ ] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [ ] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: packages/tests/server/react/abortOnUnmount.test.tsx
Comment (juliusmarminge) [inline packages/tests/server/react/abortOnUnmount.test.tsx:1]: not related to this PR but was missing so thought I'd add it cause we know trash loves tests :)

**Review by juliusmarminge — DISMISSED** [review]
👍🏼 

Let's get @KATT approval if we can deprecate the old way or if he wants to keep it in there

**Review by KATT — APPROVED** [review]

File: packages/client/src/internals/TRPCUntypedClient.ts [hunk 1]
```diff
@@ -75,6 +75,7 @@ export interface TRPCSubscriptionObserver<TValue, TError> {
 export type CreateTRPCClientOptions<TRouter extends AnyRouter> =
   | CreateTRPCClientBaseOptions<TRouter> & {
       links: TRPCLink<TRouter>[];
+      abortOnUnmount?: boolean;
```
Comment (KATT) [inline packages/client/src/internals/TRPCUntypedClient.ts:78]: Actually..... this doesn't belong in `@trpc/client` --- we don't want to have react-specific stuff there :)

**Review by KATT — CHANGES_REQUESTED** [review]

File: packages/client/src/internals/TRPCUntypedClient.ts (hunk 1 above)
Comment (bautistaaa) [reply packages/client/src/internals/TRPCUntypedClient.ts:78]: doh! good catch

**Review by KATT — APPROVED** [review]

**Pull request merged**